src/main/java/net/masterthought/cucumber/generators/AbstractPage.java

```diff
@@ -107,4 +115,20 @@ private void buildGeneralParameters() {
             }
         }
     }
+
+    /**
+     * Escapes all html and xml that was provided in a reference before inserting it into a template.
+     */
+    private final static class EscapeHtmlReference implements ReferenceInsertionEventHandler {
```

Comment:
As you see there are many tests (not only unit) that reduces number of regressions when preparing new PR.
Can you add unit test for both cases?